src/settings/OpeningPeriodForm/BigCalendarWrapper.js

```diff
@@ -53,9 +53,16 @@ class BigCalendarWrapper extends PureComponent {
         }
       }) => {
         const event = {};
+        // localized start of week
         let eventDay = moment().startOf('week').toDate();
-
-        eventDay = moment(eventDay).add(weekdays.indexOf(weekday), 'day');
+        // 0 to 6 from Sunday - Saturday.  Not localized
+        const localizedStartDay = moment(eventDay).day();
+        // distance from Sunday (where `weekdays` array starts to actual start day)
+        // add seven to keep this positive (stay in week we're displaying)
+        const localizedStartOffset = (0 - localizedStartDay) + 7;
```

Comment:
Could you also try to use global constants instead of `magick` numbers? NUMBER_OF_DAYS_IN_WEEK for example or such like this. It will be more clear for understanding when reading the code.